server/datastore/mysql/software.go

```diff
@@ -213,9 +213,9 @@ func (d *Datastore) hostSoftwareFromHostID(tx *sqlx.Tx, id uint) ([]fleet.Softwa
 	sql := `
 		SELECT s.id, s.name, s.version, s.source, coalesce(scp.cpe, "") as generated_cpe, 
 			IF(
-				JSON_ARRAYAGG(scv.cve) = JSON_ARRAYAGG(null), 
+				GROUP_CONCAT(scv.cve) = null, 
 				null, 
-				JSON_ARRAYAGG(
+				GROUP_CONCAT(
```

Comment:
Maybe this is fine, but just to make sure - this will return a string with JSON objects concatenated with a comma between each (e.g. `{"cve": "x", "details_link": ".../x"},{"cve": "y", "details_link": ".../y"}`).  Also note that any NULL instances of `scv.cve` will still create json objects with NULLs as values.

Reply:
Yeah, this is arguably not ideal, but it's caught in the `Scan` implementation: http://github.com/fleetdm/fleet/blob/f60dd23a2d7a145e733ae86ba87e0bdc5c857ea1/server/fleet/software.go#L43-L43, which we added because some versions of MariaDB don't seem to behave super nicely with JSON ops.

Reply:
Gotcha, makes sense!